espanso/Cargo.toml

```diff
@@ -55,7 +55,7 @@ simplelog = "0.9.0"
 log.workspace = true
 anyhow.workspace = true
 thiserror.workspace = true
-clap = "2.33.3"
+clap = "3.0.0"
```

Comment:
I would recommend loosening the version restriction to "3" -- no good reason to tie to a minor or bugfix release IMO.

Reply:
Also, did you look at upgrading directly to clap 4? It looks like v4 was released way back in 2022...

Reply:
Yes! I'm on my way. The upgrade guide says it first needs to be updated on 3.0, and then the latest 3.*.
Apparently there are a couple of broken apis since version 3.0.0.
I would also like to use the derive API, is _way_ less verbose and understandable that the current build API

Reply:
The guides [from v2 to v3](https://github.com/clap-rs/clap/blob/v4.4.11/CHANGELOG.md#migrating-1) and from [v3 to v4](https://github.com/clap-rs/clap/blob/v4.4.11/CHANGELOG.md#migrating)

Reply:
Interesting.

I've used the derive api in a few small personal projects -- pretty nice to work with for the 90% use case!

Reply:
I did the `"3"` change and still compiled, so I'm merging this and starting over for clap v4. Thanks again for the time!